### PR TITLE
MSC-37 Base for checking purchasability with different codes for different r…

### PR DESCRIPTION
**Pull request by tcichowicz**
…easons

### CODE REVIEW

File: test/common/expectPurchasabilityCodes.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+const { hexToNumber } = web3.utils;
+
+const expectedMaximumLength = 10;
+
+module.exports = actualCodes => ({
+  toEqual: (expectedCodes) => {
+    expect(actualCodes.length).to.equal(expectedMaximumLength);
+
+    for (let i = 0; i < expectedCodes.length; i += 1) {
```
Comment (przemyslawjanpietrzak) [inline test/common/expectPurchasabilityCodes.js]:
```
expectedCodes.forEach((actualCodes) => {
  const actualCode = hexToNumber(actualCode);
      expect(actualCode).to.equal(expectedCode);
```
is more readable

File: contracts/PurchasabilityValidator.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import "./DrugItem.sol";
+
+contract PurchasabilityValidator {
+
+    uint8 public VALID_FOR_PURCHASE = 100;
+
+    function isPurchasable(DrugItem _drugItem) public view returns (uint8[] memory) {
+        uint8[] memory result = new uint8[](10);
+        result[0] = VALID_FOR_PURCHASE;
+        return result;
```
Comment (pokuniewicz) [inline contracts/PurchasabilityValidator.sol]: you can use `public view return (uint8[] memory result)` and then skip the last line in the method ;)

File: test/common/expectPurchasabilityCodes.js (hunk 1 above)
Comment (tcichowicz) [reply test/common/expectPurchasabilityCodes.js]: Actually it's a bit more complex than that. The actual array is of fixed length but the input (expected values) is of variable length. That's why I need to check for both expected values and following zeroes. I've refactored it a bit, how do you like it now?

Comment (przemyslawjanpietrzak) [reply test/common/expectPurchasabilityCodes.js]: You're right, i thought is one array.

Hoverer still you can use .forEach method, second argument of cb is array index, so it should looks like:

```
expectedCodes.forEach((actualCodes, i) => {
  const actualCode = hexToNumber(actualCodes[i]);
      expect(actualCode).to.equal(expectedCode)
```

**Review by przemyslawjanpietrzak — APPROVED** [review]

**Pull request merged**